web/modules/custom/itk_activity/src/Form/ActivityExternalSignupUserForm.php

```diff
@@ -75,20 +77,32 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    $name = $form_state->getValue('name');
-    $email = $form_state->getValue('email');
-    $phone = $form_state->getValue('phone');
-
     $node = $form_state->get('node');
 
     if (isset($node)) {
-      $node->field_external_signed_up_users[] = $name . " - " . $email . " - " . $phone;
-
+      $name = $form_state->getValue('name');
+      $email = $form_state->getValue('email');
+      $phone = $form_state->getValue('phone');
+
+      // Create new external user entity.
+      $externalUser = ExternalUser::create([
+        'type' => 'itk_activity_external_user',
+        'name' => $name,
+        'email' => $email,
+        'phone' => $phone,
+      ]);
+      $externalUser->save();
+
+      // Update activity node.
+      $node->field_external_signed_up_users[] = $externalUser;
       $node->save();
-    }
 
-    // Add message.
-    drupal_set_message(t($name . ' is registered to activity.'));
+      // Add message.
+      drupal_set_message($name . t(' is registered to activity.'));
```

Comment:
This is not the currect usage of t(). Please see https://api.drupal.org/api/drupal/core%21includes%21bootstrap.inc/function/t/8.3.x

t($string, **array $args = []**, array $options = [])

:-)